dotnet/test/AutoGen.Tests/SemanticKernelAgentTest.cs

```diff
@@ -131,4 +132,97 @@ public async Task SemanticKernelPluginTestAsync()
+        var key = Environment.GetEnvironmentVariable("AZURE_OPENAI_API_KEY") ?? throw new Exception("Please set AZURE_OPENAI_API_KEY environment variable.");
+        var builder = Kernel.CreateBuilder()
+            .AddAzureOpenAIChatCompletion("gpt-35-turbo-16k", endpoint, key);
+
+        var kernel = builder.Build();
+
+        var connector = new SemanticKernelChatMessageContentConnector();
+        var agent = new ChatCompletionAgent() { Kernel = kernel, Name = "assistant" };
+        var skAgent = new SemanticKernelChatCompletionAgent(agent)
+            .RegisterMiddleware(connector);
+
+        var messages = new IMessage[]
+        {
+            MessageEnvelope.Create(new ChatMessageContent(AuthorRole.Assistant, "Hello")),
+            new TextMessage(Role.Assistant, "Hello", from: "user"), new MultiModalMessage(Role.Assistant,
+                [
+                    new TextMessage(Role.Assistant, "Hello", from: "user"),
+                ],
+                from: "user"),
+        };
+
+        foreach (var message in messages)
+        {
+            var reply = await skAgent.SendAsync(message);
+
+            reply.Should().BeOfType<TextMessage>();
+            reply.As<TextMessage>().From.Should().Be("assistant");
+        }
+    }
+
+    [ApiKeyFact("AZURE_OPENAI_API_KEY", "AZURE_OPENAI_ENDPOINT")]
+    public async Task SkChatCompletionAgentPluginTestAsync()
+    {
+        var endpoint = Environment.GetEnvironmentVariable("AZURE_OPENAI_ENDPOINT") ?? throw new Exception("Please set AZURE_OPENAI_ENDPOINT environment variable.");
+        var key = Environment.GetEnvironmentVariable("AZURE_OPENAI_API_KEY") ?? throw new Exception("Please set AZURE_OPENAI_API_KEY environment variable.");
+        var builder = Kernel.CreateBuilder()
+            .AddAzureOpenAIChatCompletion("gpt-35-turbo-16k", endpoint, key);
+
+        var parameters = this.GetWeatherAsyncFunctionContract.Parameters!.Select(p => new KernelParameterMetadata(p.Name!)
+        {
+            Description = p.Description,
+            DefaultValue = p.DefaultValue,
+            IsRequired = p.IsRequired,
+            ParameterType = p.ParameterType,
+        });
+        var function = KernelFunctionFactory.CreateFromMethod(this.GetWeatherAsync, this.GetWeatherAsyncFunctionContract.Name, this.GetWeatherAsyncFunctionContract.Description, parameters);
+        builder.Plugins.AddFromFunctions("plugins", [function]);
+        var kernel = builder.Build();
+
+        var agent = new ChatCompletionAgent()
+        {
+            Kernel = kernel,
+            Name = "assistant",
+            ExecutionSettings =
+                new OpenAIPromptExecutionSettings()
+                {
+                    ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions
+                }
+        };
+        var skAgent = new SemanticKernelChatCompletionAgent(agent).RegisterMiddleware(new SkSequentialChatMessageContentConnector());
```

Comment:
You can register an `IStreamingMiddleware` to an `IAgent` because `IStreamingMiddleware` is also an `IMiddleware`